### ISSUE TITLE
Search not focused on mobile when tapping search icon

### ISSUE THREAD

**Issue by Tyris** [squidfunk/mkdocs-material#5799]
### Bug description

On mobile (specifically at <960px when the search bar is replaced with just the icon), tapping on the search icon (magnifying glass) open the search pane but does not focus the input. This means that the mobile keyboard (Android and iOS) does not appear until the user taps on the search input.

Basically this means mobile uses must tap twice to begin searching.

This works fine on desktop for physical keyboards (presumably when the search pane is open all key-presses are captured), and also works fine on mobile if a zoomed out (desktop) view is used (eg: standard search input).

I'm guessing that an `input.focus()` call probably needs to be made after the search pane is displayed, but haven't done any further investigation.

### Related links

- https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/

### Reproduction

It occurs on your own help site so I'm not attaching a zip (requiring a zip is a real impediment to reporting bugs).

### Steps to reproduce

1. Go to https://squidfunk.github.io/mkdocs-material/ on mobile device.
2. Tap search
3. Keyboard doesn't appear
4. Tap the search input
5. Keyboard appears

### Browser

Chrome

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. This is something that worked for some time, and then, several years ago, Safari and other browsers prohibited delayed focusing of elements. I didn't check yet, but maybe somebody by now found a hack how to pull this off, but it will definitely be a hack.

> It occurs on your own help site so I'm not attaching a zip (requiring a zip is a real impediment to reporting bugs).

The fact that something occurs on our own site doesn't lift the need for a reproduction. Why do we need one? Because it gives us the opportunity to start working on a fix right away before setting something up. When the reproduction is minimal, working on the bug is much quicker than when it occurs on a large site, e.g., because of initial build and rebuild cycles. Thus, we require our users to add reproductions when reporting bugs. [We're closing this issue as incomplete](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#incomplete-issues).

If you attach a minimal reproduction, we're happy to look into it. Additionally, please read our [bug reporting guide](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).

**Comment by Tyris** [comment]
Thanks for the insight on the reasoning of delayed focusing - I was vaguely aware of that, but hadn't considered it in this context.

That said, with the (frankly, pretty cool) way you're handling the search transition (css driven off that hidden checkbox) I'm not sure that applies? A quick hack by adding the following at the bottom of the generated index.html seems to do the trick (though it introduces a slight visual hiccup on second and subsequent opens of the search) - at least in Chrome:

```javascript
<script>
document.getElementById("__search").addEventListener("change", (e) => {
  if (e.target.checked) {
      document.getElementsByClassName("md-search__input")[0].focus();
  }
});
</script>
```

**Comment by squidfunk** [comment]
We fixed it in 50f455ec0, at least that's what testing on iOS tells me. The fix provided in https://github.com/squidfunk/mkdocs-material/issues/5799#issuecomment-1667225851 might work on Android but it doesn't work on iOS. My research tells me that it must be triggered by a user interaction like a `click` event – `change` is not enough.
